lang/LangSource/PyrSlot64.h

```diff
@@ -107,6 +110,14 @@ inline bool NotPtr(const PyrSlot* slot) { return slot->tag != tagPtr; }
 /* setter functions */
 inline void SetInt(PyrSlot* slot, int val)           { slot->tag = tagInt;  slot->u.i = val; }
 inline void SetObject(PyrSlot* slot, struct PyrObjectHdr* val)      { slot->tag = tagObj;  slot->u.o = (struct PyrObject*)(val); }
+inline void SetNewObjectOnStack(PyrSlot* slot, NewPyrObjectPtr* val)      { slot->tag = tagObj;  slot->u.o = (struct PyrObject*)(val->release()); }
+inline void SetNewObjectInObject(PyrSlot* slot, NewPyrObjectPtr* val)
+{
+	PyrObjectHdr *obj = val->get();
+	slot->tag = tagObj;
+	slot->u.o = (struct PyrObject*)(obj);
+	val->releaseAndWriteNew(slotRawObject(slot));
+}
```

Comment:
This is a great place to use the move&& operator. If the function was instead e.g.:
`inline void SetNewObjectInObject(PyrSlot* slot, NewPyrObjectPtr&& val)``
then I believe you'd need to explicitly move e.g. `SetNewObjectInObject(obj, std::move(val))`;
This makes clear in the calling code that val is no longer valid (and, uses of val after this will probably get caught by static analysis). In turn, the GC write could also be a move (once you've done `releaseAndWriteNew`, your `NewPyrObjectPtr` is no longer "new", so it makes sense it would be consumed here)

Reply:
Okay. Can look at that.

Reply:
I was thinking that ```newPyrWhatever``` funcs should return a NewPyrObjectPtr (this would prevent circumventing the approach) but wasn't of the proper way that should work. Example code I tried to model on didn't compile. (Grr ;-) Also as this is quite central, wanted to make sure this is done as efficiently as possible. I did a bit of reading which made vague suggestions that thinks would probably maybe be very optimised, but good to be sure.

Reply:
@scztt Getting back to this finally. So just to make sure I understand, this is essentially about the semantics in the code, i.e. we're signalling that ```val``` is toast. But in this case, I don't need to actually move ```val```?

```
inline void SetNewObjectOnStack(PyrSlot* slot, NewPyrObjectPtr&& val)      { slot->tag = tagObj;  slot->u.o = (struct PyrObject*)(val.release()); }
inline void SetNewObjectInObject(PyrSlot* slot, NewPyrObjectPtr&& val)
{
	PyrObjectHdr *obj = val.get();
	slot->tag = tagObj;
	slot->u.o = (struct PyrObject*)(obj);
	val.releaseAndWriteNew(slotRawObject(slot));
}
```